File: shoebill_plugins/PlayerVehiclePlugin/src/main/java/lt/ltrp/dao/PlayerVehiclePermissionDao.java
package lt.ltrp.dao;

import lt.ltrp.constant.PlayerVehiclePermission;
import lt.ltrp.object.LtrpPlayer;
import lt.ltrp.object.PlayerVehicle;

import java.util.Collection;
import java.util.Map;

/**
 * @author Bebras
 *         2016.05.31.
 */
public interface PlayerVehiclePermissionDao {


    void remove(PlayerVehicle vehicle, int userId);
    void remove(PlayerVehicle vehicle);
    void remove(PlayerVehicle vehicle, int userId, PlayerVehiclePermission permission);
    void add(PlayerVehicle vehicle, LtrpPlayer player, PlayerVehiclePermission permission);
    void add(PlayerVehicle vehicle, int userId, PlayerVehiclePermission permission);
    void add(int vehicleId, int userId, PlayerVehiclePermission permission);
    Collection<PlayerVehiclePermission> get(int vehicleId, int userId);
    Map<Integer, PlayerVehiclePermission> get(int vehicleId);

}

File: shoebill_plugins/PlayerVehiclePlugin/src/main/java/lt/ltrp/player/vehicle/dao/PlayerVehiclePermissionDao.kt
package lt.ltrp.player.vehicle.dao

import lt.ltrp.player.vehicle.constant.PlayerVehiclePermission
import lt.ltrp.player.`object`.LtrpPlayer
import lt.ltrp.player.vehicle.`object`.PlayerVehicle

/**
 * @author Bebras
 * *         2016.05.31.
 */
interface PlayerVehiclePermissionDao {


    fun insert()

    fun remove(vehicle: PlayerVehicle, userId: Int)
    fun remove(vehicle: PlayerVehicle)
    fun remove(vehicle: PlayerVehicle, userId: Int, permission: PlayerVehiclePermission)
    fun add(vehicle: PlayerVehicle, player: LtrpPlayer, permission: PlayerVehiclePermission)
    fun add(vehicle: PlayerVehicle, userId: Int, permission: PlayerVehiclePermission)
    fun add(vehicleId: Int, userId: Int, permission: PlayerVehiclePermission)
    operator fun get(vehicleId: Int, userId: Int): Collection<PlayerVehiclePermission>
    operator fun get(vehicleId: Int): Map<Int, PlayerVehiclePermission>

}
